### ISSUE TITLE
KeyError for `added` in build_file_change_history.py

### ISSUE THREAD

**Issue by penyuan** [OPEN-NEXT/wp2.2_dev#29]
**How to reproduce:** Run current `master` branch `start.py` with the repository `jurgenwesterhoff/bGeigieNanoKit`.

**Symptom:** Run fails with this traceback:

```
retrieving commits in jurgenwesterhoff/bGeigieNanoKit
fetching info at https://github.com/jurgenwesterhoff/bGeigieNanoKit
Traceback (most recent call last):
  File "/home/penyuan/.vscode-oss/extensions/ms-python.python-2020.5.86806/pythonFiles/ptvsd_launcher.py", line 48, in <module>
    main(ptvsdArgs)
  File "/home/penyuan/.vscode-oss/extensions/ms-python.python-2020.5.86806/pythonFiles/lib/python/old_ptvsd/ptvsd/__main__.py", line 432, in main
    run()
  File "/home/penyuan/.vscode-oss/extensions/ms-python.python-2020.5.86806/pythonFiles/lib/python/old_ptvsd/ptvsd/__main__.py", line 316, in run_file
    runpy.run_path(target, run_name='__main__')
  File "/home/penyuan/.pyenv/versions/3.6.10/lib/python3.6/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/home/penyuan/.pyenv/versions/3.6.10/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/home/penyuan/.pyenv/versions/3.6.10/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/penyuan/devel/git/wp2.2_dev/src/start.py", line 174, in <module>
    main()
  File "/home/penyuan/devel/git/wp2.2_dev/src/start.py", line 139, in main
    build_file_change_history(known_commits, file_change_history)
  File "/home/penyuan/devel/git/wp2.2_dev/src/build_file_change_history.py", line 47, in build_file_change_history
    added = filechange["added"],
KeyError: 'added'
fish: “env PYTHONIOENCODING=UTF-8 PYTH…” terminated by signal SIGTERM (Polite quit request)
```

Looks like sometimes the metadata retrieved with Perceval might not include the `added` item.

**Comment by jbon** [comment]
For some reason, while most of dict elements in the list of dict "files" in a commit contain the keys "action", "added" and "removed", some of them don't. I fixed the bug created by this "inconsistency" of the data in 91c7134. Instead of blindly trying to fetch the missing information, we now test first if it is there and therewith avoid the KeyErr. If the info is not there, a warning is raised. 

Tested successfully with:
- OPEN-NEXT,WP2.2_reference
- jbon,github-mining
- jurgenwesterhoff,bGeigieNanoKit

This fix will allow us to continue working without this bug appearing. Nonetheless, it is only half satisfactory: we should investigate why the data is "inconsistent". I'll create a new issue for this.



@penyuan: from my side, ready to be reviewed.
